### PR TITLE
shuffle oc commands into logical package structure

### PR DESCRIPTION
**Pull request by deads2k**
`pkg/oc` is the result of snipping a bunch of links to the main repo without regard to final resting place.  This updates it to look more like

1. pkg/oc/cli - all the commands go here
2. pkg/oc/clusteradd - still needs love.  We really need to split cobra from logic
3. pkg/oc/clusterup - still needs love.  We really need to split cobra from logic
4. pkg/oc/lib - the old idea of having a library of function that you can use given clients and input (ancient briant grant issue).  it's still a good idea. Things in here shouldn't have cobra or flag deps and are things we want to keep(ish)
5. pkg/oc/originpolymorphichelpers - separate.  We want this to be purely external one day and potentially put it in kube
6. pkg/oc/util - well, every family has the crazy uncle.  These we want to die.

/assign @soltysh @juanvallejo @mfojtik

### CODE REVIEW

**Comment by deads2k** [comment]
oh, that's not bad at all.

File: pkg/oc/cli/cmd/example.go [hunk 1]
```diff
@@ -1,80 +0,0 @@
-package cmd
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/example.go]: any reason for removing this file altogether?

File: pkg/oc/cli/extract/extract.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package cmd
+package extract
```
Comment (juanvallejo) [inline pkg/oc/cli/extract/extract.go]: why not `pkg/oc/cli/cmd/extract/extract.go`?

File: pkg/oc/cli/idle/idle.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package cmd
+package idle
```
Comment (juanvallejo) [inline pkg/oc/cli/idle/idle.go]: same comment as above

File: pkg/oc/cli/idle/idle_test.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package cmd
+package idle
```
Comment (juanvallejo) [inline pkg/oc/cli/idle/idle_test.go]: same comment as above

File: pkg/oc/cli/importer/appjson/appjson.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package importer
+package appjson
```
Comment (juanvallejo) [inline pkg/oc/cli/importer/appjson/appjson.go]: same as above, why not move within the `cmd` dir?

File: pkg/oc/cli/cmd/example.go (hunk 1 above)
Comment (deads2k) [reply pkg/oc/cli/cmd/example.go]: > any reason for removing this file altogether?

it was dead.

File: pkg/oc/cli/newbuild/newbuild.go [hunk 6]
```diff
@@ -224,7 +225,7 @@ func (o *BuildOptions) RunNewBuild() error {
 	return nil
 }
 
-func transformBuildError(err error, baseName, commandName, commandPath string, groups errorGroups, config *newcmd.AppConfig) {
+func transformBuildError(err error, baseName, commandName, commandPath string, groups ocnewapp.ErrorGroups, config *newcmd.AppConfig) {
```
Comment (mfojtik) [inline pkg/oc/cli/newbuild/newbuild.go]: should ErrorGroups go into util?

Comment (deads2k) [reply pkg/oc/cli/newbuild/newbuild.go]: > should ErrorGroups go into util?

They appear to be a peculiarity of newapp.

File: pkg/oc/cli/cmd/example.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/oc/cli/cmd/example.go]: Oh, I thought we kept it as a template to guide new contributors?

**Comment by danwinship** [comment]
@deads2k this appears to have broken `./hack-update-deps.sh` because `tools/testdebug/cmd/load_etcd.go` still refers to `pkg/oc/admin/policy` which no longer exists

**Comment by danwinship** [comment]
(There are also references to `pkg/oc/admin/policy` in a few comments, and in `hack/import-restrictions.json`)

**Pull request merged**